File: tests/xcalablemp/global-view/gmove/C/error/gmove_bca_2a1t_c.c
#include <stdio.h>
#include <xmp.h>

int n=8;
double a[n][n],b[n][n];
#pragma xmp nodes p(2,2)
#pragma xmp template tx(0:n-1,0:n-1)
#pragma xmp distribute tx(cyclic,cyclic) onto p
#pragma xmp align a[i][j] with tx(i,j)
#pragma xmp align b[i][*] with tx(i,*)

int main(){

  int i,j,ierr;
  int myrank;

  myrank=xmp_all_node_num();

#pragma xmp loop (i,j) on tx(i,j)
  for(i=0;i<n;i++){
    for(j=0;j<n;j++){
      a[i][j]=i+j+2;
    }
  }

#pragma xmp loop (i) on tx(i,*)
  for(i=0;i<n;i++){
    for(j=0;j<n;j++){
      b[i][j]=0;
    }
  }

#pragma xmp gmove
  b[4:2][4:1]=a[4:2][4:1];

  ierr=0;
#pragma xmp loop (i) on tx(i,*)
  for(i=4;i<5;i++){
    for(j=4;j<5;j++){
      printf("i=%d,j=%d,b=%f,myrank=%d\n",i,j,b[i][j],myrank);
    }
  }

  printf("max error=%d\n",ierr);

  return 0;

}

File: tests/xcalablemp/global-view/gmove/Fortran/error/gmove_bca_2a1t_c.f90
program tgmove
integer i,n=8
integer a(n,n),b(n,n)
integer xmp_node_num
!$xmp nodes p(2,2)
!$xmp template tx(n,n)
!$xmp distribute tx(cyclic,cyclic) onto p
!$xmp align a(i,j) with tx(i,j)
!$xmp align b(i,*) with tx(i,*)

!$xmp loop (i,j) on tx(i,j)
do j=1,n
  do i=1,n
    a(i,j)=i+j
  end do
end do

do j=1,n
!$xmp loop (i) on tx(i,*)
  do i=1,n
    b(i,j)=0
  end do
end do

!$xmp gmove
b(1:2,1:2)=a(1:2,1:2)

ierr=0
do j=1,2
!$xmp loop (i) on tx(i,*)
  do i=1,2
    ierr=ierr+abs(b(i,j)-i-j)
  end do
end do

!$xmp reduction (max:ierr)
irank=xmp_node_num()
if (irank==1) then
  print *, 'max error=',ierr
endif
!call exit(ierr)

stop
end
